plugins/modules/accesspoint_workflow_manager.py

```diff
@@ -0,0 +1,690 @@
+            self.log("Unable to get device info "+ str(e) , "ERROR")
+
+    # Below function used to show the current state of the network divices
+    # As Accesss point Device info, Config and Final field to pass in the Update data
+    def get_have(self):
+        """
+        This function used to get AP device details as json response from DNAC site.
+        by giving MAC address as a input in the URL GET Method
+        parameters:
+          key = ap_ethernet_mac_address from the device list
+        returns:
+          This will return the filtered data.
+        Device information given in the input file
+        Device Configuration Details 
+        Conpare with input data and current config data
+        """
+        responses = {}
+        ap_data = []
+        final_ap_data = []
+        ap_data = self.get_ap_configuration()
+        final_ap_data = self.compare_ap_cofig_with_inputdata(ap_data)
+        # If the final_ap_data is None means no changes required to update
+        if final_ap_data is None:
+            responses["accesspoint"] = dict(
+                device_list= self.payload["device_list"],
+                accesspoint_config=ap_data,
+                final_input=final_ap_data,
+                msg="Input Access Point Configuration remain same in Current AP configration")
+            del self.payload["device_list"]
+            self.result["skipped"] = True
+        else:
+            responses["accesspoint"] = {
+                "device_list": self.payload["device_list"],
+                "device_configs": ap_data,
+                "final_input": final_ap_data,
+                "msg": "Filtered AP Device data list for update AP config"}
+        self.result["response"].append(responses)
+        return final_ap_data
+
+    def get_ap_configuration(self):
+        """
+        This function used to get AP device details as json response from DNAC site.
+        by giving MAC address as a input in the URL GET Method
+        Parameters:
+            We are not using input param already passed in the module.param
+          - self.payload: used from yml input files input.yml we are using accesspoints section
+            if ap_selected_field in input.yml is empty or all this will show all field
+            else ap_selected_field: "mac_address,eth_mac,ap_name,led_status,location"
+            then will show only listed field.
+        Returns:
+            [
+                {
+                    "apName": "NFW-AP1-9130AXE",
+                    "ethMac": "34:5d:a8:0e:20:b4",
+                    "ledBrightnessLevel": 3,
+                    "ledStatus": "Enabled",
+                    "location": "LTTS",
+                    "macAddress": "90:e9:5e:03:f3:40"
+                }
+            ]
+        Example:
+            functions = Accesspoint(module)
+            ap_data = functions.get_ap_configuration()
+        """
+        ap_config_data = []
+        for device in self.payload["device_list"]:
```

Comment:
What if device_list is not there it will give keyError so better to use self.payload.get("device_list") in order to avoid key error exception.